File: backend/src/models/Organisation.model.js
import mongoose from 'mongoose';

const Schema = mongoose.Schema;

const organisationSchema = new Schema({
  name: { type: String, required: true },
  type: { type: String, enum: ['school', 'college'], required: true },
  domain: { type: String, sparse: true, unique: true },
  address: { type: String },
  contactEmail: { type: String },
  contactPhone: { type: String },
  subscription: {
    plan: { type: String, enum: ['free', 'premium', 'enterprise'], default: 'free' },
    validUntil: { type: Date },
    maxTeachers: { type: Number, default: 10 },
    maxStudents: { type: Number, default: 500 }
  },
  status: { type: String, enum: ['active', 'suspended', 'trial'], default: 'trial' },
  createdBy: { type: Schema.Types.ObjectId, ref: 'User' },
}, { 
  timestamps: true 
});

organisationSchema.index({ status: 1 });

export default mongoose.model('Organisation', organisationSchema);

File: frontend/lib/shared/models/organisation_model.dart
import 'package:equatable/equatable.dart';

class OrganisationModel extends Equatable {
  final String id;
  final String name;
  final String type; // 'school', 'college'
  final String? domain;
  final String? address;
  final String? contactEmail;
  final String? contactPhone;
  final String status; // 'active', 'suspended', 'trial'

  const OrganisationModel({
    required this.id,
    required this.name,
    required this.type,
    this.domain,
    this.address,
    this.contactEmail,
    this.contactPhone,
    required this.status,
  });

  factory OrganisationModel.fromJson(Map<String, dynamic> json) {
    final normalized = _normalizeJson(json);
    return OrganisationModel(
      id: normalized['id'] as String,
      name: normalized['name'] as String,
      type: normalized['type'] as String,
      domain: normalized['domain'] as String?,
      address: normalized['address'] as String?,
      contactEmail: normalized['contactEmail'] as String?,
      contactPhone: normalized['contactPhone'] as String?,
      status: normalized['status'] as String? ?? 'trial',
    );
  }

  static Map<String, dynamic> _normalizeJson(Map<String, dynamic> json) {
    return {
      ...json,
      'id': (json['id'] ?? json['_id'] ?? '').toString(),
      'contactEmail': json['contactEmail'] ?? json['email'] ?? json['contact_email'],
      'contactPhone': json['contactPhone'] ?? json['phone'] ?? json['contact_phone'],
    };
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'type': type,
      'domain': domain,
      'address': address,
      'contactEmail': contactEmail,
      'contactPhone': contactPhone,
      'status': status,
    };
  }

  @override
  List<Object?> get props => [id, name, type, domain, address, contactEmail, contactPhone, status];
}
